traits/trait_list_object.py

```diff
@@ -124,6 +127,9 @@ def __init__(self, trait, object, name, value):
     def _send_trait_items_event(self, name, event, items_event=None):
         """ Send a TraitListEvent to the owning object if there is one.
         """
+        if not self.is_root_container:
+            return
+
```

Comment:
Have you considered this:
```
object = self.object()
if not getattr(object, self.name) is self:
    return
```

Reply:
This is good! Tried it, works if the root is a `List` or `Dict` or `Set`, but doesn't if the container was inside something like `Either` or `Union` `Either(Set(), None)`.